mtags/src/main/scala/scala/meta/internal/semver/SemVer.scala

```diff
@@ -1,12 +1,22 @@
 package scala.meta.internal.semver
 
+import scala.util.Try
+
 object SemVer {
 
-  case class Version(major: Int, minor: Int, patch: Int) {
+  case class Version(
+      major: Int,
+      minor: Int,
+      patch: Int,
+      releaseCandidate: Int,
+      milestone: Int
```

Comment:
Do you think it's best to model this as an `Int`? Would an `Option[Int]` not be more appropriate?

Reply:
I tried with option but that became rather complex, since I had to include the scenario where we compared release candidate with milestone. Though this will not work currently, so I might need to rethink it and do it with option.

Reply:
3.0.0 will be lower than 3.0.0-M1 currently, which I didn't think of before 😱

Reply:
Added option, aaaaand a lot more tests haha.